sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py

```diff
@@ -27,6 +31,57 @@
 JSON = MutableMapping[str, Any]  # pylint: disable=unsubscriptable-object
 T = TypeVar("T")
 ClsType = Optional[Callable[[PipelineResponse[HttpRequest, HttpResponse], T, Dict[str, Any]], Any]]
+PollingReturnType_co = TypeVar("PollingReturnType_co", covariant=True)
+_FINISHED = frozenset(["succeeded", "canceled", "failed", "completed"])
+
+
+def _parse_operation_id(operation_location_header):
+    regex = "[^:]+://[^/]+/documentintelligence/.+/([^?/]+)"
+    return re.match(regex, operation_location_header).group(1)
+
+
+class AnalyzeDocumentLROPollingMethod(LROBasePolling):
+    @property
+    def operation_id(self) -> str:
+        """Analyze operation result ID.
+
+        :rtype:str
+        """
+        return _parse_operation_id(self._initial_response.http_response.headers["Operation-Location"])
+
+
+class AnalyzeDocumentLROPoller(LROPoller[PollingReturnType_co]):
```

Comment:
Also need to override `from_continuation_token` on the custom poller here for the same reasons

Reply:
Added